include/kernels/staggered_quark_smearing.cuh

```diff
@@ -0,0 +1,229 @@
+#pragma once
+
+#include <dslash_helper.cuh>
+#include <color_spinor_field_order.h>
+#include <gauge_field_order.h>
+#include <color_spinor.h>
+#include <dslash_helper.cuh>
+#include <index_helper.cuh>
+#include <kernels/dslash_pack.cuh> // for the packing kernel
+
+namespace quda
+{
+
+  /**
+     @brief Parameter structure for driving the covariatnt derivative operator
+  */
+  template <typename Float, int nSpin_, int nColor_, int nDim, QudaReconstructType reconstruct_>
+  struct StaggeredQSmearArg : DslashArg<Float, nDim> {
+    static constexpr int nColor = 3;
+    static constexpr int nSpin  = 1;
+    static constexpr bool spin_project = false;
+    static constexpr bool spinor_direct_load = false; // false means texture load
+    typedef typename colorspinor_mapper<Float, nSpin, nColor, spin_project, spinor_direct_load>::type F;
```

Comment:
style note: prefer the use of `using` over `typedef`.

Reply:
fixed